\chapter{Introduction}

\section{Background}

\section{The promise of single-cell biology}
Biology is a watery, messy soup. The cell, mostly a bag of water filled with a generous portion of biological macromolecules like proteins and nucleic acids, is a hub of continuous, seemingly random, riotous motion \cite{berg1993random}. This randomness, while seemingly chaotic at the microscopic level, is statistically reproducible on the cellular and organismal scale, enabling life to reproduce itself. Ensembles of proteins and cells work in organized unison to create functional and successful organisms, such as the one reading this thesis. A key question in biology is how these ensembles work.

We have some reasonable answers to this question. In human biology, for instance, there are around 20,000 genes in the genome. Each cell translates a subset of these genes into proteins, and the identity of these proteins determines a cell's form and function. Broadly, the differential presence and abundance of these proteins and their post-translational modifications are what make a neuron, for instance, different from a blood cell. As far as we know, blood cells do not become neurons, even though they have the genes available. Instead, cell types maintain their identities. During development, these cellular identities are established early and are often maintained throughout the life of the organism. While this model is broadly explanatory, the long-standing goals of biology include understanding which molecules are expressed in cells, when this expression occurs, and how it is maintained.

To return to the bag metaphor: how does this messy bag of water and biomolecules create an organism comprising thousands of distinct bags working in autonomous unison? This question is fascinating in and of itself, but the benefits of developing a deep understanding of development are immediately practical. A real understanding of these molecular processes would allow us to program biology towards the goal of eliminating human diseases. A major target for such reprogramming efforts is the human immune system, which is in a constant state of development and diversification.

Recently, single-cell sequencing technologies have been developed that allow us to define the molecular processes at work in cells by comprehensively measuring which molecules are present. In this thesis, I used these technologies to build a quantitative understanding of the diversity of cell types and proteins in the human immune system. I pursued this goal in three related projects.

The first project aimed to create a comprehensive parts-list for human biology. Specifically, we abstracted the cellular bag as containing just RNA \cite{quake2021cell}: what's in the bags? The second project measured the maintenance of B cell identities and bias during cellular reprogramming: if the contents of the bag must change, how is the identity of the bag maintained? The third project aimed to understand the statistics that describe and govern the dynamics of the human B cell repertoire: what governs the decisions of the bags to change their composition and location in the body?


\section{A Parts-List for Human Biology}
In the first part of my thesis, I collaborated with the Tabula Sapiens consortium to capitalize on recent advances in our ability to systematically measure the identities of macromolecules in single cells. This parts-list approach to biology allows for a systematic understanding of the various functions cells can perform. To use an analogy, if an organism were a kitchen, we are now able to classify all the appliances within by enumerating the parts in each. We now understand what parts make up the toasters, refrigerators, and blenders of biology. Specifically, I employed recent advances in single-cell RNA sequencing techniques to quantify gene expression and cellular identities at the individual cell level \cite{klein_droplet_2015, macosko2015highly}, an ideal level of abstraction for understanding tissue and organismal biology. As I detail in this chapter, we created one of the broadest single-cell references in existence \cite{tabula_sapiens_consortium_tabula_2022}, complemented by other more focused efforts \cite{dominguez_conde_cross-tissue_2022}.

\section{Lineage Tracing during Cell Reprogramming}
In the second part of my thesis, I used B cells as a model to quantitatively understand the maintenance of cell-intrinsic identities as cells differentiate. Phil Hodgkin and his associates have made significant contributions to our understanding of immune cell development and function through the application of lineage tracing techniques \cite{hodgkin2012cell, marchingo2014t}. With the advent of immune repertoire sequencing and single-cell RNA sequencing, researchers can now perform high-resolution clonal lineage tracing, linking cellular differentiation processes to gene expression profiles of lineages of cells \cite{stubbington2017t, horns2020memory}. In this work, I quantify the strong intrinsic fate biases of different types of B cells to the same stimulus, describe the extent to which transcriptional memory is passed through generations of B cells, and characterize the magnitude of this effect in Bone-Marrow Derived Plasma Cells \textit{in vivo}.

\section{Lineage Tracing the B Cell Repertoire across Tissues}
In the third part of my thesis, I collaborated with Ivana Cvijovic on a project to map the statistics of the human B cell repertoire. These statistics prove useful for building models of how B cell differentiation proceeds in humans. This work builds on the work in Chapter 2 by generating a quantitative single-cell genomics dataset for B cell repertoire analysis. We procured immune-rich organs (spleen, lymph nodes, blood, and bone marrow) from otherwise immunologically healthy individuals and processed these cells via single-cell RNA sequencing. Using the lineage information contained in the antibody sequence, we were able to reconstruct lineage histories, and thus the human B cell differentiation process.

\section{Thesis Overview}
The chapters of this thesis focus on research I conducted on similar but distinct topics, as outlined above. Each chapter is arranged to stand as its own document. Much of the work in each chapter is published (Chapters 2 and 3) or will soon be published (Chapters 4 and 5).