% Full title as you would like it to appear on the page
\chapter{Lineage tracing reveals fate bias and transcriptional memory in human B cells}
% Short title that appears in the header of pages within the chapter
\chaptermark{Lineage Tracing \textit{in vitro}}

\section{Abstract}
We combined single-cell transcriptomics and lineage tracing to understand fate choice in human B cells. Using the antibody sequences of B cells, we tracked clones during \textit{in vitro} differentiation. Clonal analysis revealed a subset of IgM+ B cells which were more proliferative than other B cell types. Whereas the population of B cells adopted diverse states during differentiation, clones had a restricted set of fates available to them; there were two times more single-fate clones than expected given population-level cell-type diversity. This implicated a molecular memory of initial cell states that was propagated through differentiation. We then identified the genes which had strongest coherence within clones. These genes significantly overlapped known B cell fate determination programs, suggesting the genes which determine cell identity are most robustly controlled on a clonal level. Persistent clonal identities were also observed in human plasma cells from bone marrow, indicating that these transcriptional programs maintain long-term cell identities \textit{in vivo}. Thus, we show how cell-intrinsic fate bias influences differentiation outcomes \textit{in vitro} and \textit{in vivo}. 

\section{Introduction}
A key focus of developmental biology is the relationship between the molecular milieu of a progenitor cell and its differentiation outcomes. These outcomes are variously referred to as cell fate, cell identity, or cell state. Lineage tracing offers a powerful way to map which progenitor cells adopt which cell fates. Even rudimentary cell labeling techniques show clonally related offspring are biased toward similar cell fates\cite{whitman_embryology_1878}, and recent technological advances confirm the same with greater throughput and resolution. However, the contribution of cell-extrinsic versus cell-intrinsic molecular factors as determinants of cell fates remains largely uncharacterized.

To better understand cell fate determination, multiple groups have used high-throughput sequencing to measure endogenous or transgenic DNA barcodes as labels of cellular lineage\cite{lu_tracking_2011, naik_diverse_2013}. Recently, it is possible to use high-throughput sequencing to perform both lineage tracing and transcriptomics in single cells. This combination directly measures the molecular relationships between progenitors and their offspring, allowing stronger inference of molecular determinants of cell fates\cite{biddy_single-cell_2018, ludwig_lineage_2019, weinreb_lineage_2020}. For example, by analyzing the transcriptomes of lineages biased towards efficient reprogramming outcomes, Biddy et al were able to identify a previously uncharacterized methyltransferase which increased stem-cell reprogramming efficiency by threefold.

In the human immune system, clonal lineages of leukocytes rapidly proliferate while adopting diverse cell fates. This dynamic occurs \textit{in vivo} as a response to varied pathogenic challenges such as viruses, bacteria, or cancer. Spatially organized cellular structures, called germinal centers, orchestrate this process \textit{in vivo}. However, \textit{in vitro} differentiation protocols using only T or B cells can recapitulate important features of the germinal center, and provide valuable insight into the process\cite{deenick_switching_1999}. \textit{in vitro}, a researcher can control most extrinsic factors, such as cell density, cytokine cocktails, and media compositions, allowing them to study cell-intrinsic differentiation programs. In the case of B and T cells, well-controlled extrinsic conditions still reliably generate a large diversity of cell fates, indicating a strong contribution of intrinsic cell diversity to population-level diversity seen \textit{in vivo}\cite{cheon_cyton2_2021}. The question of how gene expression responds to extracellular stimuli, while a cell maintains its identity, is generally poorly understood. And the transcriptional programs underlying cell-intrinsic clonal fate bias remain largely uncharacterized. Furthermore, how extrinsic signals, intrinsic state, and clonal population structure interact to determine the dynamics of the B cell immune response remains poorly understood.

Here, we gain insight into some of these gaps in knowledge by obtaining lineage and single-cell RNA-sequencing measurements of differentiating human B cells. We used the B cell receptor (BCR) gene as a genetic lineage marker and we paired this information with a transcriptomic readout of cellular identity during an \textit{in vitro} differentiation of human B cells. These multi-modal data allow us to quantitatively infer the intrinsic biases B cells have towards specific cell fates, analyze the clonal dynamics of \textit{in vitro} B cell activation, and characterize transcriptional memory both \textit{in vitro} and \textit{in vivo}.

\section{Pilot Experiments}

To determine the feasibility and utility of lineage tracing and single-cell RNA sequencing, we used Smart-Seq2 to analyze the differentiation and fate choice of human B cells, we stimulated primary naïve B cells with cytokines \textit{in vitro} and performed single-cell RNA-sequencing  (Figure \ref{fig:paper2_prelimfig_1}A). We purified naïve B cells from peripheral blood mononuclear cells (PBMCs) via negative selection against other hematopoetic markers by magnetic cell sorting. Purified cells were 96 \% CD19+ and 97 \% IgD+. We dyed the cells with a division tracking dye called CellTrace Yellow which allowed us to assess their proliferation. The cytokine combination that we used for stimulation, CD40L, IL-2, IL4, and IL21, was shown to induce proliferation, class switch recombination, and differentiation into memory B cells or antibody-secreting cells\cite{konforte_il-21_2009}. After 3, 7, 10, and 14 days, we sorted individual cells into well-plates and performed full-length single cell RNA-sequencing. As a baseline for comparison, we also sorted and sequenced individual purified naive B cells. After quality filtering, we obtained the transcriptomes of 986 stimulated B cells and 364 naive B cells for further analysis.

\subsubsection{Differentiating Naive B cells have diverse transcriptional states}

Dimensionality reduction by principal components analysis (PCA) and UMAP\cite{mcinnes_umap_2018} of single cell transcriptomes revealed several transcriptionally distinct states, which we annotated based on known and novel markers  (Figure \ref{fig:paper2_prelimfig_1}B). Unstimulated cells were distinguished by their high expression of known naïve B cell markers such as IGHD and TSC22D3. The Early-Activation cluster was characterized by expression of secreted cytokines such as lymphotoxin-alpha (LTA) and interleukin-6 (IL6), as well as transcription factor BATF3. GC-ready cells were distinguished by expression of BCL6 and TNFAIP3. We also identified a cluster of cells which appeared to be in the act of class-switch recombination (CSR). We called them switching cells because they expressed AICDA and many of the cells in the cluster had switched classes based on splicing analysis  (Figure \ref{fig:paper2_prelimfig_1}C). Furthermore, the transcriptional program of these cells was consistent with the G1 stage of the cell cycle, in line with reports that CSR occurs in G1\cite{abbott2016germinal}. SERF2 distinguished these cells. SERF2 is gene previously not described in B cell activation, but this gene may reflect a hypoxia-induced metabolic program known to be associated with class-switching\cite{abbott2016germinal}. Finally, expression of PRDM1, a known plasma cell marker, separated plasmablasts from the rest of the activated B cells  (Figure \ref{fig:paper2_prelimfig_1}C).

\subsection{Clonally related B cells have more similar transcriptomes than unrelated cells}
B cell receptor sequences can serve as markers for the clonal origins of individual B cells. Using the full-length transcriptome sequencing data, we computationally assembled the B cell receptor of each single cell and assigned cells to clones based on their sequences. Among the 986 stimulated B cells, 475 cells (48\%) were clonally related to at least one other cell. These cells belonged to 145 clones. In contrast, the unstimulated naïve B cells were clonally unrelated to all other cells in the dataset, as expected based on the exceptionally high diversity of naïve B cells in the human repertoire\cite{briney2019commonality}.

We used the lineage information to find that clonally related B cells have more similar transcriptomes than unrelated cells. Clones tended to cluster together in the UMAP embeddings. We calculated the Euclidean distance between cells in the UMAP and PCA space(Figure \ref{fig:paper2_prelimfig_2})B, which showed the transcriptional distance between related cells was generally smaller than between unrelated cells. This calculation is strongly caveated by the fact that linear UMAP distance is non-intuitive to interpret. In the subsequent section of my thesis, I show a more principled calculation of transcriptome level similarity based on a neighborhood graph clustering metric. To resolve individual cell division events within clones, we examined the division-tracking dye\cite{hasbold1998cell}. We fit a gaussian mixture model to our division data, which allowed us to assign peaks to divisions TODO. We used each cell’s division number to understand the extent of clonally inherited gene expression across cell divisions. Interestingly, over the course of 4 divisions, transcriptional distance increased, perhaps consistent with that large amount of genes induced during the differentiation program, then later in the division-time transcriptional distances between clones fell to below a null-expection  (Figure \ref{fig:paper2_prelimfig_2}C).  

\subsection{Persistence of transcriptional memory varies across loci}
Clonal correlations of transcriptome state are underpinned by the persistence of transcriptional memory at individual genetic loci. In principle, some genes may exhibit faithful transmission of transcriptional state across cell divisions, while other genes do not. Thus, we sought to determine how persistence of transcriptional memory varies across the genome.
We adopted a statistical approach, described in detail in the methods section to find clonally regulated genes and measure their persistence. We identified a set of 65 genes with Benjamini-Hochberg corrected p-values $<$ 0.01, which we call clonally coherent genes. Supporting the sensitivity of our test, we identified genes known to be clonally inherited: the heavy-chain variable genes, the light chain variable genes, and the light chain constant regions. 

\subsection{Non-coding switch transcription exhibits strongly persistent memory}

Non-coding transcription of the IgH constant regions genes, called switch transcription here, is thought to direct CSR by targeting AID to intronic switch regions  (Figure \ref{fig:paper2_prelimfig_3})A\cite{stavnezer_igh_2014}. For this reason, we examined the switch transcription of clones of B cells at the IgH locus. Clones had strikingly concordant switch transcription  (Figure \ref{fig:paper2_prelimfig_3})B, indicating the robust inheritance of a local transcriptional state over multiple cell divisions, and a mechanism whereby B cell clones switch to the same isotype during the immune response. We used the cell tracking dye to understand how persistent these switch transcription memories were. We found correlations between clones remained high through at least 5 divisions before diminishing to a level expected by random chance  (Figure \ref{fig:paper2_prelimfig_3})C.

\subsection{Clonal Gene Expression states can be explained by variable Progenitor States}
The strong concordance within lineages at the IgH locus was often driven by IGHE: lineages either tended to express IGHE highly or none at all  (Figure \ref{fig:paper2_prelimfig_3}D) This effect was reflected in our statistical tests where IGHE was one of the top 10 clonal genes. IGHE expression is induced by IL4 signaling, yet the B cells were activated in-vitro with the same amount of IL4 in culture. Thus, we hypothesized some Naive B cells are intrinsically primed toward IL4-sensitivity. In line with this hypothesis, we detect IL4R expression in 20 \% of Naïve B cells from the same donor (Figure \ref{fig:paper2_prelimfig_3}D). In further support of inheritance of IL4 signaling, we also identify IL4I (IL4 induced) as a clonal gene (p $<$ 0.001).

\section{Conclusions from preliminary experiments}

\subsection{Biological Conclusions}
While we observed a diversity of phenotypes in the population of activated B cells, we identified clonally inherited transcriptional modules which were stable over the course of many cell divisions, demonstrating clonal similarity in a background of highly diverse activation states. Among these inherited expression states were cell-type defining genes such as MS4A1, HLA-DQB, and IGHE, in addition to genes which were not yet described in B cell-biology. We observed tight concordance in transcription state between lineages at the IgH locus, which was clear in first divisions and persisted over time-scales relevant for B cell fate choices\cite{hodgkin_modifying_2018}. This concordance suggests an inherited chromatin state which is faithfully propagated over multiple cell divisions. Thus, B cell clones may become intrinsically and stably biased towards a particular effector fate. As combining lineage tracing with single cell genomics becomes more common, we suggest clonal genes can be used for feature selection in single cell RNA-seq data. In future lineage tracing studies, we expect to uncover more connections between the cellular states of progenitor cells and the gene expression states of their differentiating clonal lineages.

\subsection{Limitations and Scaling Up Single Cell Transcriptomics}
Our analysis and conclusions were hampered by the size and amount of clones detected, as well as the inability to track the same population of cells over a time course. Additionally, our pilot experiments used different experimental conditions for different replicates, and we saw large amounts of variability in \textit{in vitro} This is largely due to the lower-throughput of Smart-seq2\cite{baran2018experimental}. Given our interesting results, we reattempted the experiments using   10X Genomics 5' system are two widely used single-cell RNA sequencing (scRNA-seq) technologies that differ in their methodology and performance characteristics. Smart-seq2, developed and published by Picelli et al. in 2014\cite{picelli2014full}, is a full-length mRNA sequencing method that captures and amplifies the entire transcript. This approach generates high-quality data with a broader coverage of gene expression, enabling better detection of low-abundance transcripts and alternative splicing events. However, Smart-seq2 is labor-intensive and costly when scaling to large numbers of cells.

In contrast, the 10X Genomics 5' system, which gained prominence in 2018, is a droplet-based scRNA-seq method that captures the 5' end of the mRNA. This technique uses barcoded beads in a microfluidic device to encapsulate individual cells, enabling the high-throughput processing of thousands of cells simultaneously. The 10X Genomics 5' system sacrifices full-length transcript coverage for scalability, making it more suitable for large-scale studies. Critically, this system also allows researchers to sequence the 5' end of transcripts, where the VDJ portion of the antibody is contained. This allows immune profiling, and lineage tracing on the scale of 100,000s of cells, as demonstrated in the following sections and chapters. 

\section{Results}

\subsection{In vitro–differentiated B cells recapitulate major aspects of \textit{in vivo} B cell development}

To study differentiation and fate choice of human B cells, we performed an \textit{in vitro} differentiation protocol using primary human B cells from healthy donors. We simulated T cell–dependent activation of B cells using a cocktail of cytokines (CD40L, IL2, IL4, IL10, and IL21); (see the Methods note 1 section). These cytokines induced proliferation, class-switch recombination (CSR), and reprogramming into terminally differentiated Plasma cells. We performed single-cell RNA-sequencing on time course samples from this protocol(Figure \ref{fig:paper2_fig_1}1A), which furnished us with population-genetic and transcriptional information about B cell differentiation. In addition, we contextualized our \textit{in vitro} differentiation by (1) integrating our single-cell RNA-sequencing data with publicly available data from 10X Genomics and (2) measuring CD138+ bone marrow plasma cells from a separate donor (i.e., terminally differentiated B cells \textit{in vivo}). After quality control and bioinformatic exclusion of non-B cells, we obtained the mRNA transcriptome and VDJ sequences for 29,703 B cells from our six samples(Figure \ref{fig:paper2_fig_1}B and C).
Dimensionality reduction by principal component analysis and UMAP\cite{mcinnes_umap_2018} of the single-cell transcriptomes revealed several distinct cell states. We automatically annotated cell states with CellTypist (Figure \ref{fig:paper2_fig_1}D)\cite{dominguez_conde_cross-tissue_2022}, and more granularly based on multi-modal information we collected about the BCR (Figure \ref{fig:paper2_fig_s1}S1C). We quantified the relative abundances of these algorithmically determined cell states over time, and found dramatic changes (Figure \ref{fig:paper2_fig_1}E). First, we noted that non-B cells present in the input rapidly became undetectable by day 4 (Figure \ref{fig:paper2_fig_s1}S1D), which shows the specificity of the cytokines for B cell expansion. Other notable dynamics included a threefold decrease in the relative abundance of plasma cells from Day 0 to 4 and a 1,000-fold increase in Proliferative Germinal center B cells. Finally, we observed substantial amounts of cell death (30\%) by Day 4, implicating cell death as a major contributor to the population dynamics (Figure \ref{fig:paper2_fig_s1}S1E).


\subsection{Measuring VDJ mutation status allows inference of population-level cell-fate biases}

Upon experiencing antigenic stimulation, naive B cells genetically diversify their BCR by accruing mutations in their germline VDJ genes (somatic hypermutation), as well as switching expression of constant region genes through DNA deletion events (CSR). These endogenous processes have been used to make lineage inferences \textit{in vivo}\cite{horns_lineage_2016}. We reasoned we could use these endogenous genetic alterations in the BCR as estimators of the initial cell state of any given cell detected \textit{in vitro}. For example, we infer a B cell with an unmutated BCR detected in the time course likely arose from a naive B cell progenitor.

We assessed the validity of this approach by quantifying the concordance between transcriptionally defined memory and naive B cell states and categorically delineated somatic hypermutation levels (germline, mutated, heavily mutated). In general, the concordance between mutational and transcriptionally defined cell state categories was high (Figure \ref{fig:paper2_fig_s2}S2A–C). For example, in the Day 0 population, more than 97 \% naive B cells possessed an unmutated BCR gene, also known as a germline gene, and less than 11 \% of plasma cells were classified germline. There was also a continuum of transcriptional states which we labeled “B cells,” which encompasses a gradient of cell identities between switched-memory B cells and naive B cells. Critically, we observed no appreciable evidence of hypermutation in our \textit{in vitro} conditions (Figure \ref{fig:paper2_fig_s2}S2D), consistent with prior literature showing BCR stimulus is necessary for \textit{in vitro} hypermutation\cite{bergthorsdottir_signals_2001}.


Given the germline and mutated categories classified naive and naive transcriptional states in the input population with high specificity, we used the hypermutation level measured for \textit{in vitro} differentiated B cells as a confident prediction of whether their progenitor cell was a naive or naive B cell. We found the progeny of hypermutated B cells increased twofold in relative abundance over the course of the culture, showing hypermutated B cells are intrinsically twofold more persistent \textit{in vitro}, in our conditions (Figure \ref{fig:paper2_fig_2}2A). This is consistent with orthogonal measurements, which report memory B cells are on average one division ahead of naive B cells when cultured \textit{in vitro}\cite{tangye_intrinsic_2003}.


We next analyzed how the mutation status was related to their transcriptional identity (Figure \ref{fig:paper2_fig_2}2B). On Day 0, we found what we expected in the peripheral blood. For example, transcriptionally identified plasma cells were four times more often mutated than germline. However, by Day 4, germline cells populated the plasmablast/cell state almost as often as mutated cells, definitively linking naive B cell progenitors to plasmablast phenotypes in this culture system, showing that circulating naive B cells can rapidly adopt a plasmablast-like phenotype. As the differentiation proceeded, mutated B cells began to repopulate the plasmablast/cell compartment, suggesting most naive B cell–derived antigen-secreting cells are short-lived.

We continued to use the mutation status of the IgH locus as a lineage marker, which allowed us to understand the population dynamics of cell states \textit{in vitro}. To better understand differences in the early activation programs of mutated and germline B cells, we analyzed the differentially expressed genes (DEGs) between these subsets of activated B cells (i.e., mutated versus germline Proliferative Germinal Center B cells). We found mutated B cells were likely to express genes involved in T-cell interaction such as CD70, CCL17, and CCL21 (Figure \ref{fig:paper2_fig_2}(Fig 2C). It is known that memory B cells are intrinsically licensed to enter an inflammatory state which activates T cells\cite{liu_memory_1995, good_resting_2009}, and our results describe the gene expression program which orchestrates this propensity. In contrast, germline B cells were biased toward expressing SELL, CLEC2B, and proliferative markers, suggesting naive B cells are intrinsically primed to home into the lymphatic system and proliferate in germinal centers.

Our measurement of phenotypes was not limited to the transcriptome because B cells generated additional phenotypic diversity \textit{in vitro} through CSR. Generally, as naive IGHM+ B cells experience cytokinetic/antigenic stimulation, they class-switch to any of the IGHA, IGHG, or IGHE genes. This process diversifies the immune response by producing antibodies with the same specificity, but different effector functions. We quantified the \textit{in vitro} dynamics of CSR through the lens of mutation status, which revealed strongly different fate biases between germline and mutated cells (Figure \ref{fig:paper2_fig_2}D). Most strikingly, B cells which switched to IGHE were almost exclusively derived from germline progenitors: the ratio of germline IGHE cells to mutated IGHE cells was (eightfold - inf, 95 \% CI). In  (Figure \ref{fig:paper2_fig_2}E), we illustrate models of cell state biases which were calculated from our population lineage tracing.

\subsection{Clonal analysis reveals clonal fate bias, MZ-like B cells, and a map of class-switch events
}
We next used the full BCR sequence to identify clones in our dataset. Clones were defined as having identical CDR3s and using the same heavy-chain V gene. Among the 11,333 differentiating B cells, 1,911 were clonally related to at least one other detected cell (Figure \ref{fig:paper2_fig_3}A S3A). Using the paired clonal and transcriptomic information, we determined that clones had very strong fate biases (Figure \ref{fig:paper2_fig_3}B). In this analysis, we defined fates by Leiden clustering (Figure \ref{fig:paper2_fig_s3}(Fig S3B). Clones were twofold more likely to be found in a single fate, than expected given the large diversity of transcriptional states in the population and controlling for variation such as mutation status or sequencing batch. Only about 5 \% of clones adopted more than two fates, showing that although multi-potency was possible, strong fate biases within clones were the norm.

We detected 73 clones with family members detected at Day 0 and at a later time point (Figure \ref{fig:paper2_fig_s3}(Fig S3C). We called these persistent clones. Among persistent clones, IGHM+ B cells with mutated VDJs (B cells\_mutated\_IGHM) was the most common progenitor state at Day 0. This suggests B cells with this phenotype are hyperproliferative compared with other B cells, as has been observed by others\cite{seifert_functional_2015}. To test this hypothesis on our data, we modeled a scenario in which persistence was equal amongst all Day 0 clones: division rates were the same, and death rates were zero. We detected about 2X more progeny of this cell state than would be expected in the case of the aforementioned even-expansion model (Figure \ref{fig:paper2_fig_3}(Fig 3C). We wondered whether these persistent clones had a more granular identity, which was not detected by the standard single-cell clustering and differential expression approaches used to annotate them in the first place. To this end, we performed differential expression analysis on the persistent clones versus non-persistent clones found within the “B cells\_mutated\_IGHM” subset. This analysis revealed persistent clones were characterized by high expression of CD1C, FTX, and LPP. These clones also had low expression of TAX1BP1 and CD27 (Figure \ref{fig:paper2_fig_3}(Fig 3D). We describe these cells as MZ-like B cells because their phenotype resembles circulating Splenic marginal zone B cells, which have also been shown to respond rapidly to immune challenges\cite{weller_human_2004}. These cells also bear a resemblance to age-associated B cells (ABCs)\cite{cancro_age-associated_2020}.

We also used the clonal information to understand the \textit{in vitro} dynamics of CSR. On the population level, we observed an order of magnitude increase in the amount of class-switched cells above the input (Figure \ref{fig:paper2_fig_s3}(Fig S3C). We used the observed intraclonal isotype counts to derive a map of class-switch outcomes \textit{in vitro} (Figure \ref{fig:paper2_fig_3} (Figs 3E and S3E). For comparison, we calculated the naive probability of detecting a switch given the proportions of isotype usage in the general population. For same–same isotype relationships (i.e., IGHG1 IGHG1), the map of \textit{in vitro} class-switching showed more than 10-fold enrichment compared with the naive probability model. This enrichment can be explained by clonal inheritance of isotype status. We also noted a strong divergence from this model for the IGHM to IGHA1 switch. Although the naive probability model expects a large amount of IGHM to IGHA1 switches, we detected few. These data show CSR from IGHM cells did not meaningfully contribute to the abundance of IGHA+ cells in the population, as would be expected given a lack of TGF-$\beta$ in the cocktail\cite{stavnezer_surprising_2009}. Thus, our clonal analysis definitively clarified whether IGHA+ cells in the output came from the differentiation process or through proliferation of existing IGHA+ cells. In contrast, we noted that many intraclonal class-switching events appeared to be directly from IGHM to IGHE, showing that direct switching was more probable than sequential switching in our conditions. Given this IGHE+ cells are generally not detected in the peripheral blood \textit{in vivo}, there are likely efficient intrinsic or niche-based factors which limit the appearance of these IGHE+ cells in the peripheral blood during an immune response.


\subsection{Persistence of transcriptional memory varies across genetic loci}
The intrinsic biases in fate outcomes that we detected must be underpinned by the persistence of transcriptional memory at individual genetic loci. In principle, some genes may exhibit faithful transmission of transcriptional state across clonal expansion, whereas other genes may not. Thus, we sought to determine how the persistence of transcriptional memory varies across the genome. We used a permutation test on the clone labels to find genes which were less variable within clones than between.

Using this test, we identified a set of 6,937 genes with P-values less than < 0.01. Supporting the sensitivity of our test, we identified genes known to be clonally inherited: the light chain variable genes and the light chain constant regions. These genes were not used per se to identify clones, providing clear evidence for the validity of the test. We call the genes we identified clonally coherent genes (CCGs). One way to intuitively understand these clonal effects is to observe the cascade plot of the putative CCG, for example, IGKC  (Figure \ref{fig:paper2_fig_4}A (Horton et al, 2018). A clear clonal structure of high expressing clones and low expressing clones is obliterated upon permuting clonal labels. Whether we detected a CCG was highly related to its expression level. For lowly expressed genes, we often could not reject the null hypothesis, likely because their measured expression levels are dominated by technical noise such as dropout (Figure \ref{fig:paper2_fig_4}(Fig 4B).

We calculated an effect size of the variability in gene expression explained by the clonal labels called the clonal index (see the Materials and Methods section). Unsurprisingly, the Ig variable genes and light chain genes had some of the strongest effects (Figure \ref{fig:paper2_fig_4}(Fig 4C). These values provide helpful calibration for how strong clonal effects are in other genes. In contrast to the light chain, where cells generally only transcribed one constant region gene, we measured robust transcription from multiple different heavy-chain constant region genes in single cells. This transcription is consistent with the so-called sterile transcription which is necessary for CSR (Lee et al, 2001). Strikingly, we found Ig heavy-chain constant region genes such as IGHE were CCGs (Figure \ref{fig:paper2_fig_s4}(Fig S4A), which was surprising given the diversity of transcriptional states measured for the locus. The quantitative clonal coherence of IgH transcription suggests faithful propagation of chromatin states at the IgH locus across cell division, and may be an explanation for observations of clonal coherence in isotype usage \textit{in vivo} (Horns et al, 2016).

We noticed that the CCGs with the strongest effects were often genes known B cell fates, such as MS4A1, AIDCA, and JCHAIN. To quantify this observation, we calculated the overlap coefficient between the set of top CCGs and top DEGs between B cell states. We observed strong agreement between the set of top DEGs and top CCGs (0.39 overlap), which was 14-fold higher than the agreement between DEGs and null-sets of genes sampled from the B cell transcriptome (Figure \ref{fig:paper2_fig_4}D). Using a set of important B cell genes curated from the literature\cite{morgan_unraveling_2022}, we also found an eightfold enrichment for CCGs. These enrichments show the CCGs are involved in cell fate determination, are relevant features for functional characterization, and could be used for feature selection in single-cell workflows versus highly variable genes.

Finally, we asked whether we could identify CCGs from \textit{in vivo} samples. We used a cascade plot to visualize the clonal expression levels of IGKC+ bone marrow plasma cells. We observed intraclonal coherence in these samples as well, where certain clones were IGKC high, and others were IGKC low expressers (Figure \ref{fig:paper2_fig_s4}A). When we tested the entire transcriptome we found hundreds of CCGs, which is an order of magnitude fewer genes than the \textit{in vitro} data (Figure \ref{fig:paper2_fig_s4}B). Nonetheless, the genes we identified were important for plasma cell function such as MZB1, JCHAIN, and the IgH constant region genes. JCHAIN was detected as a CCG both \textit{in vitro} and \textit{in vivo}. Thus, we speculate that differentiating B cell clones make a stable fate choice to express whether JCHAIN. Given CD38+CD138+ Plasma cells could have been generated by immune responses years in the past\cite{hammarlund_plasma_2017}, we expect we are detecting inherited transcriptional programs on the scale of years later.

\section{Conclusions}

An effective immune response requires profound and rapid differentiation while maintaining tight control of cell identity. We combined single-cell genomics and lineage tracing to investigate the differentiation process of human primary B cells. Using this multi-modal measurement, we observed a large diversity of phenotypes in differentiating B cells. In the population of activated B cells, we were able to explain much of this diversity through inference of progenitor states. Populations of memory and naive B cells showed clear differences in their proliferative ability and the gene expression programs they adopted in response to the same stimulus. Importantly, typical single-cell RNA-seq measurements would not resolve this meaningful heterogeneity. At a more granular level, our clonal analysis also allowed us to characterize MZ-like B cells circulating in the blood, which again would be difficult to identify via standard single-cell RNA-sequencing techniques because of their rarity and subtle phenotypic differences on purely transcriptional level. Interestingly, these cells had very low expression of TAX1BP1. This low expression may be a molecular explanation for their high proliferation rates and proclivity towards plasma cell differentiation, as has been seen in TAX1BP1 knockout cell lines\cite{matsushita_regulation_2016}.

We found that B cell clones were likely to choose a single-cell fate, even in the midst of the many possible fates given the diverse cytokine stimulus. This fact holistically demonstrates how intrinsic molecular heterogeneity in the starting population is an important contributor to population diversity in immune responses, as has been demonstrated for specific loci\cite{wu_intrinsic_2017}. These molecular heterogeneities are often seen in high-dimensional data, but their significance has heretofore been unclear. Although we used B cells and cytokines as a model, the cell-intrinsic effects described herein operate in other scenarios, such as cancer drug resistance\cite{goyal_pre-determined_2021}. To systematically understand these clonal effects, we characterized the transcriptomic similarity of clones. We detected CCGs in our \textit{in vitro} samples, as well as in plasma cells isolated directly from human bone marrow. The set of CCGs was heavily enriched for immune response genes such as JCHAIN, MS4A1, and AIDCA, while simultaneously containing hundreds of genes which are likely relevant features of B cell biology, but not yet investigated. Taken together, our data show how intrinsic clonal transcriptional identities are faithfully propagated during cell differentiation \textit{in vitro} and \textit{in vivo}. Our multi-modal molecular measurements were crucial for characterizing these effects, and should allow more useful probabilistic models of the immune system\cite{hodgkin_modifying_2018}. These models of the immune system will benefit from continued systematic efforts to compile molecular information about cell states\cite{tabula_sapiens_consortium_tabula_2022}.

Here we exploited the unique biology of B cells to gain insight into their differentiation processes. We conceptualized the BCR as an \textit{in vivo} molecular recorder of antigen or germinal center experience. This allowed us to dissect the differences in activation programs between the progeny of memory and naive B cells. These differences are particularly interesting given the progeny would not be distinguishable via typical single-cell genomics workflows. We speculate similar intrinsic heterogeneity and cell fate biases exist in other immune cells\cite{sanmiguel_hand_2020}, but have yet to be fully described because of a lack of detailed clone tracking. In general, as lineage tracing and cell-recording technologies continue to develop, researchers will genetically record bespoke cell experiences, and use single-cell genomics to analyze their effects on intrinsic cell identities\cite{chen_connecting_2022}. Finally, we note that \textit{in vitro} differentiation has moved out of research labs and into the clinic in the form of CAR-T therapies and regenerative medicine efforts. Single-cell genomics already offers powerful insight into cell therapeutics\cite{bode_exploiting_2021}, but adding \textit{in vivo} and \textit{in vitro} lineage tracing technologies will yield critical additional power to characterize rare cells, improve targeting, and increase potency.


%%%%%%%%%%%%%
%% Paper 1 %%
%%%%%%%%%%%%%
\begin{figure}[hbt!]
\centering
\includegraphics[width=8cm, keepaspectratio]{figs/prelim_paper2/BCellLineagePaper_Figure 1.png}
\caption[Experimental overview and transcriptional data for Naive B cell activation].{(A) Experimental Scheme (B) Annoated UMAP of the identified transcriptional states (C) UMAPs of marker gene expression levels and inferred CSR status. (D) Violin plots of marker genes for each cell state}
\label{fig:paper2_prelimfig_1}
\end{figure}

\begin{figure}[htb!]
\centering
\includegraphics[width=8cm, keepaspectratio]{figs/prelim_paper2/BCellLineagePaper_Figure 2.png}
\caption[Transcriptomic similarity of clones \textit{in vitro}]{(A) UMAP projection showing the largest clones in the dataset (top) and the dye tracking division number of each clone (bottom) (B) Transcriptional distances between related and unrelated cells in UMAP (top) and PCA (bottom) space. p$<$0.01 by Kolmogorov-Smirnov test (C) UMAP distance plotted by division sum (2 divsions are siblings, 4 divisions are cousins, etc) of the related pairs of observed cells}
\label{fig:paper2_prelimfig_2}
\end{figure}

\begin{figure}[hbt!]
\centering
\includegraphics[width=14cm, keepaspectratio]{figs/prelim_paper2/BCellLineagePaper_Figure 3.png}
\caption[Concordant Switch Transcription States amongst related cells]{(A) A schematic of the IgH locus and switch transcription (B) Point plots showing diverse yet concordant switch transcription states amongst clonal groups (C) Boxen plot showing concordance in switch transcription state decreased over division-time (D) Cascade plot showing clonal effects in IGHE transcription levels}
\label{fig:paper2_prelimfig_3}
\end{figure}


%%%%%%%%%%%%%
%% Paper 2 %%
%%%%%%%%%%%%%

\begin{figure}[hbt!]
\centering
\includegraphics[width=12cm, keepaspectratio]{figs/paper2/fig1_bcd.png}
\caption[Experimental overview for studying \textit{in vitro} B cell dynamics using integrated single-cell genomics and lineage tracing.]{(1) B cells or Plasma cells purified from blood or bone marrow using MACS. (2) B cells from the same purification stimulated with the StemCell B cell Expansion kit. \textit{in vitro} differentiation samples collected on days 4, 8, and 12. (3) Single-cell genomic data collected and analyzed schematic. (B) UMAP embedding separate cells into distinct clusters, with dots as cells colored by sample origin. (B, C) Countplot of B cells passing QC for each sample (colors same as (B)). (D) UMAP embedding with cell type annotations. (E) Proportion of B cell types in each sample. Time course samples are connected by lines. Error bars represent 95 percentile intervals calculated by resampling.}
\label{fig:paper2_fig_1}
\end{figure}

\begin{figure}[hbt!]
\centering
\includegraphics[width=14cm, keepaspectratio]{figs/paper2/figs1_bcd.jpg}
\caption[Analysis of stimulation cocktail and presentation of pre-processing steps.]{(A) Identities and relative abundance over the media of a panel of 80 human cytokines. Cytokines with fold change $>$ 2 are shown. SN, supernatant. PBS, Phosphate Buffered Saline. (B) UMAPs of all cells in the dataset, colored as shown in legends. The Celltypist-generated labels were kept for downstream analysis, except “Age-associated B cells” were changed to “B cells.” (C) UMAPs colored by multi-modal labeling of B cells. (D) Pointplot quantifying the proportion of celltypes in each sample\_id. (E) Scatter plots of representative FACS data. (F) Top differentially expressed genes between cell types}
\label{fig:paper2_fig_s1}
\end{figure}

\begin{figure}[hbt!]
\centering
\includegraphics[width=14cm, keepaspectratio]{figs/paper2/fig2_bcd.png}
\caption[Characterization of cell-intrinsic phenotypes using VDJ mutation status.]{(A) Proportion of germline, mutated, and heavily mutated B cells in the samples; gray boxes illustrate BM CD138+ which is not part of the time course but serves as a comparison sample. (B) Ratio of germline to mutated cells in each cell state over the time course. (C) Top differentially expressed genes in between mutation categories in Proliferative germinal center B cells. Gene expression is log base 2 umis per 10,000. (D) Ratio of germline to mutated cells for each isotype group. Gray box as in (A). (B, E) Illustration of the inferred cell-type biases from (B). Error bars in all figures are 95 perecentile confidence intervals calculated by resampling.}
\label{fig:paper2_fig_2}
\end{figure}


\begin{figure}[hbt!]
\centering
\includegraphics[width=14cm, keepaspectratio]{figs/paper2/figs2_bcd.jpg}
\caption[Validation of population based lineage inference.]{(A) Empirical cumulative distributions of the fraction of V-gene based mutated away from the germline V-gene, for each transcriptomically defined B cell type. (A, B) Heat map (confusion matrix) showing concordance between the mutation status based on (A) and each B cell type label based on Leiden clustering. (A, C) UMAP plot colored by the mutation status assigned based on (A). (D) Empirical cumulative distributions of the SD in V-identity within clones showing mutations are not collected within the V-genes of clones over the time course.}
\label{fig:paper2_fig_s2}
\end{figure}

\begin{figure}[hbt!]
\centering
\includegraphics[width=14cm, keepaspectratio]{figs/paper2/fig3_bcd.jpg}
\caption[Clonal families allow inference of intrinsic proliferative ability, limited clonal fate outcomes, map of class-switching \textit{in vitro}.]{(A) UMAP embedding displays the largest clonal families detected. (B) Stacked histogram compares cell fate outcomes within clones to permuted clonal labels, with permutation restricted within respective mutational groups to analyze intrinsic bias not explainable by mutation status. (C) B cells with mutated IGHM BCRs show over-representation in the differentiated population. Observed data contrasted with a model where Day 0 clonal structure expands evenly for eight divisions and is randomly resampled. (D) Differential expression analysis between persistent and non-persistent clones in mutated IGHM B cell population. (E) Observed-to-expected clonal isotype relationship ratio for detected clonal isotype relationships.}
\label{fig:paper2_fig_3}
\end{figure}

\begin{figure}[hbt!]
\centering
\includegraphics[width=14cm, keepaspectratio]{figs/paper2/figs3_bcd.jpg}
\caption[Clonal analysis of cell identity outcomes.]{(A) Clone size distribution of the B cell population from the \textit{in vitro} time course. (B) UMAP showing results of the Leiden clustering used for clonal fate bias calculations. (C) UMAP showing all cells from persistent clones in black. Persistent clones are clones detected at multiple time points. (D) Class-switching (isotype) dynamics during the culture. ND, not detected in the Day 0 population, a pseudo-count of 1 is added for visualization purposes. (E) Countplots of the largest clones detected and their transcriptomically defined cell fates and amplicon sequencing defined isotype status.}
\label{fig:paper2_fig_s3}
\end{figure}

\begin{figure}[hbt!]
\centering
\includegraphics[width=14cm, keepaspectratio]{figs/paper2/fig4_bcd.png}
\caption[Clonal transcriptional programs are strongly enriched for fate determining genes.]{(A) Prototypical cascade plots showing IGKC expression amongst IGKC+ clones determined via immune repertoire sequencing. Each column is a clonal family of size $\geq$ 4. Families are rank-ordered by the mean gene expression of the family. Each dot is a cell, colored by the cell state label. The true clonal families are plotted on the left and a permutation of the clonal labels is plotted on the right. (B) Boxen plot showing the distribution of \% dropout for all genes tested. Genes across the entire distribution of detection rates were detected as clonal genes. Genes not detected as clonal were often very lowly expressed. (C) A volcano plot showing results of the transcriptome-wide permutation test. Q values are the Benjamini–Hochberg–corrected P-values and the clonal index is a normalized metric of expression variance described in the Materials and Methods section. Genes of interest are labeled and groups of interest are colored. Known clonal genes are the variable immunoglobulin genes. (D) The set of top CCGs strongly overlap with the set of top cell state defining genes compared to sets of randomly selected genes (P < 0.001). The null expectation is the B cell transcriptome, which were size-matched sets of genes sampled randomly from the set of genes in at least 10 \% of B cells.}
\label{fig:paper2_fig_4}
\end{figure}


\begin{figure}[hbt!]
\centering
\includegraphics[width=13cm, keepaspectratio]{figs/paper2/figs4_bcd.jpg}
\caption[Analysis of persistent \textit{in vivo} transcriptional programs.]{(A) (Top) Cascade plot of IGHE expression in differentiating B cell clones, colored by the isotype associated with their VDJ sequence. (Bottom) Cascade plot of IGKC expression of IGKC+ clones in the bone marrow sample (B) A volcano plot showing results of the transcriptome-wide permutation test only for the \textit{in vivo} (BM CD138+) sample. The q-values are Benjamini–Hochberg–corrected and the clonal index is a normalized metric of expression variance described in the Materials and Methods. Genes of interest are labeled and groups of interest are colored. (C) ecdf plot of the number of clonal genes detected with a q-value $<$ 0.01 by the fraction of cells expressing a given gene (also known as \% gene dropout).}
\label{fig:paper2_fig_s4}
\end{figure}


